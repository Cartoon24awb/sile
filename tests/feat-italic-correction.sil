\begin[papersize=a6]{document}
\nofolios
\neverindent
\font[size=16pt]
\use[module=packages.math]
% Italic correction.
% The setting plays at paragraph level
% Disabled:
\set[parameter=typesetter.italicCorrection, value=false]
− (\em{fluff}) ¡\em{fancyful proof}! [\em{puff}] \em{fluff}²

% Enabled:
\set[parameter=typesetter.italicCorrection, value=true]
+ (\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]  \em{fluff}²

% The setting also works in hbox-built content.
% We want to be sure the hbox width is still properly computed
% Hence the | marks.
% Disabled:
\set[parameter=typesetter.italicCorrection, value=false]
− |\hbox{(\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]}|

% Enabled:
\set[parameter=typesetter.italicCorrection, value=true]
+ |\hbox{(\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]}|

% Extra check with French exceptions.
\language[main=fr]{%
% Disabled:
\set[parameter=typesetter.italicCorrection, value=false]
− «\em{fluff}» \em{french proof}! (\em{fluff})

% Enabled:
\set[parameter=typesetter.italicCorrection, value=true]
+ «\em{fluff}» \em{french proof}! (\em{fluff})
}

% Repeated with another font
\font[family=Cormorant Infant]
\set[parameter=typesetter.italicCorrection, value=false]
− (\em{fluff}) ¡\em{fancyful proof}! [\em{puff}] \em{fluff}²

\set[parameter=typesetter.italicCorrection, value=true]
+ (\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]  \em{fluff}²

\set[parameter=typesetter.italicCorrection, value=false]
− |\hbox{(\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]}|

\set[parameter=typesetter.italicCorrection, value=true]
+ |\hbox{(\em{fluff}) ¡\em{fancyful proof}! [\em{puff}]}|

\language[main=fr]{%
\set[parameter=typesetter.italicCorrection, value=false]
− «\em{fluff}» \em{french proof}! (\em{fluff})

\set[parameter=typesetter.italicCorrection, value=true]
+ «\em{fluff}» \em{french proof}! (\em{fluff})
}

\end{document}
