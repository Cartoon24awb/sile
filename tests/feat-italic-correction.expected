Set paper size 	297.6377985	419.5275636
Begin page
Mx 	14.8819
My 	33.4764
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	2149 w=6.5625	(−)
Mx 	25.6818
T	11 w=5.0703	(()
Mx 	30.7521
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	55.5958
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	64.9035
T	163 w=4.3516	(¡)
Mx 	69.2551
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 68 81 70 92 73 88 79 w=49.0313	(fancyful)
Mx 	122.1385
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	153.2244
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	161.8134
T	62 w=4.9219	([)
Mx 	166.7352
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	83 88 747 w=24.2578	(puff)
Mx 	190.9930
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	64 w=4.9219	(])
Mx 	200.1523
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	224.9960
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	2011 w=5.2578	(²)
Mx 	14.8819
My 	52.6764
T	14 w=6.5625	(+)
Mx 	25.6796
T	11 w=5.0703	(()
Mx 	31.2573
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	58.7886
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	68.0941
T	163 w=4.3516	(¡)
Mx 	73.2269
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 68 81 70 92 73 88 79 w=49.0313	(fancyful)
Mx 	126.1084
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	159.8111
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	168.3979
T	62 w=4.9219	([)
Mx 	173.3511
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	83 88 747 w=24.2578	(puff)
Mx 	200.2965
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	64 w=4.9219	(])
Mx 	209.9932
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	237.2433
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	2011 w=5.2578	(²)
Mx 	14.8819
My 	71.8764
T	2149 w=6.5625	(−)
Mx 	25.6864
T	95 w=3.9063	(|)
Mx 	29.5927
T	11 w=5.0703	(()
Mx 	34.6630
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	59.5067
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	68.8191
T	163 w=4.3516	(¡)
Mx 	73.1707
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 68 81 70 92 73 88 79 w=49.0313	(fancyful)
Mx 	126.0583
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	157.1443
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	165.7379
T	62 w=4.9219	([)
Mx 	170.6597
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	83 88 747 w=24.2578	(puff)
Mx 	194.9176
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	64 w=4.9219	(])
Mx 	199.8394
T	95 w=3.9063	(|)
Mx 	14.8819
My 	91.0764
T	14 w=6.5625	(+)
Mx 	25.6848
T	95 w=3.9063	(|)
Mx 	29.5910
T	11 w=5.0703	(()
Mx 	35.1688
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	62.7000
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	72.0107
T	163 w=4.3516	(¡)
Mx 	77.1435
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 68 81 70 92 73 88 79 w=49.0313	(fancyful)
Mx 	130.0297
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	163.7324
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	172.3244
T	62 w=4.9219	([)
Mx 	177.2777
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	83 88 747 w=24.2578	(puff)
Mx 	204.2230
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	64 w=4.9219	(])
Mx 	209.1449
T	95 w=3.9063	(|)
Mx 	14.8819
My 	110.2764
T	2149 w=6.5625	(−)
Mx 	25.6878
T	169 w=7.7656	(«)
Mx 	36.8405
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	65.0713
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	170 w=7.7656	(»)
Mx 	77.0803
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 85 72 81 70 75 w=38.3594	(french)
Mx 	119.2974
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	152.4974
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	161.0923
T	11 w=5.0703	(()
Mx 	166.1626
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	191.0064
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	14.8819
My 	129.4764
T	14 w=6.5625	(+)
Mx 	25.6872
T	169 w=7.7656	(«)
Mx 	36.8398
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	65.0704
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	170 w=7.7656	(»)
Mx 	77.0515
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	73 85 72 81 70 75 w=38.3594	(french)
Mx 	119.2680
T	83 85 82 82 73 w=31.0859	(proof)
Mx 	152.4680
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	4 w=4.3516	(!)
Mx 	161.0624
T	11 w=5.0703	(()
Mx 	166.6401
Set font 	Gentium Plus;16;400;Italic;normal;;;LTR
T	739 88 747 w=24.8438	(fluff)
Mx 	194.1714
Set font 	Gentium Plus;16;400;;normal;;;LTR
T	12 w=5.0703	())
Mx 	14.8819
My 	148.6764
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2531 w=6.3680	(−)
Mx 	25.7546
T	2403 w=4.9440	(()
Mx 	30.6986
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	54.4266
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
Mx 	63.8754
T	2372 w=3.9520	(¡)
Mx 	67.8274
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	3.8880
Mx 	6.9440
Mx 	7.5360
Mx 	5.5200
Mx 	7.2000
Mx 	4.1920
Mx 	7.4400
Mx 	3.7920
T	680 a=3.8880 457 a=6.9440 567 a=7.4400 486 a=5.5200 665 a=6.9760 517 a=4.1920 634 a=7.4400 555 a=3.7920	(fancyful)
Mx 	118.8441
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	155.8928
T	2407 w=4.3840	([)
Mx 	160.2768
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	604 634 816 w=22.0960	(puff)
Mx 	182.3728
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2408 w=4.3840	(])
Mx 	191.2616
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	214.9896
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2328 w=4.0160	(²)
Mx 	14.8819
My 	167.8764
T	2530 w=6.3680	(+)
Mx 	25.7524
T	2403 w=4.9440	(()
Mx 	31.4242
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	57.2482
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
Mx 	66.6947
T	2372 w=3.9520	(¡)
Mx 	72.3427
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	3.8880
Mx 	6.9440
Mx 	7.5360
Mx 	5.5200
Mx 	7.2000
Mx 	4.1920
Mx 	7.4400
Mx 	3.7920
T	680 a=3.8880 457 a=6.9440 567 a=7.4400 486 a=5.5200 665 a=6.9760 517 a=4.1920 634 a=7.4400 555 a=3.7920	(fancyful)
Mx 	123.3573
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Mx 	153.7652
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	162.2198
T	2407 w=4.3840	([)
Mx 	167.0311
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	604 634 816 w=22.0960	(puff)
Mx 	191.2202
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2408 w=4.3840	(])
Mx 	200.8283
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	226.6523
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2328 w=4.0160	(²)
Mx 	14.8819
My 	187.0764
T	2531 w=6.3680	(−)
Mx 	25.7597
T	2627 w=2.6880	(|)
Mx 	28.4477
T	2403 w=4.9440	(()
Mx 	33.3917
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	57.1197
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
Mx 	66.5734
T	2372 w=3.9520	(¡)
Mx 	70.5254
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	3.8880
Mx 	6.9440
Mx 	7.5360
Mx 	5.5200
Mx 	7.2000
Mx 	4.1920
Mx 	7.4400
Mx 	3.7920
T	680 a=3.8880 457 a=6.9440 567 a=7.4400 486 a=5.5200 665 a=6.9760 517 a=4.1920 634 a=7.4400 555 a=3.7920	(fancyful)
Mx 	121.5472
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	158.6009
T	2407 w=4.3840	([)
Mx 	162.9849
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	604 634 816 w=22.0960	(puff)
Mx 	185.0809
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2408 w=4.3840	(])
Mx 	189.4649
T	2627 w=2.6880	(|)
Mx 	14.8819
My 	206.2764
T	2530 w=6.3680	(+)
Mx 	25.7580
T	2627 w=2.6880	(|)
Mx 	28.4460
T	2403 w=4.9440	(()
Mx 	34.1177
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	59.9417
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
Mx 	69.3938
T	2372 w=3.9520	(¡)
Mx 	75.0418
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	3.8880
Mx 	6.9440
Mx 	7.5360
Mx 	5.5200
Mx 	7.2000
Mx 	4.1920
Mx 	7.4400
Mx 	3.7920
T	680 a=3.8880 457 a=6.9440 567 a=7.4400 486 a=5.5200 665 a=6.9760 517 a=4.1920 634 a=7.4400 555 a=3.7920	(fancyful)
Mx 	126.0619
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Mx 	156.4699
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	164.9300
T	2407 w=4.3840	([)
Mx 	169.7413
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	604 634 816 w=22.0960	(puff)
Mx 	193.9304
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2408 w=4.3840	(])
Mx 	198.3144
T	2627 w=2.6880	(|)
Mx 	14.8819
My 	225.4764
T	2531 w=6.3680	(−)
Mx 	25.7604
T	2441 w=7.2960	(«)
Mx 	36.6559
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	63.9835
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2442 w=7.2960	(»)
Mx 	75.7900
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	4.1920
Mx 	5.1520
Mx 	5.4880
Mx 	7.5360
Mx 	5.5200
Mx 	7.1200
T	517 a=4.1920 607 a=5.4400 498 a=5.4880 567 a=7.4400 486 a=5.5200 527 a=7.1200	(french)
Mx 	115.3085
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Mx 	146.1469
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	154.6094
T	2403 w=4.9440	(()
Mx 	159.5534
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	183.2814
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
Mx 	14.8819
My 	244.6764
T	2530 w=6.3680	(+)
Mx 	25.7599
T	2441 w=7.2960	(«)
Mx 	36.6553
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	63.9827
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2442 w=7.2960	(»)
Mx 	75.5482
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
Mx 	4.1920
Mx 	5.1520
Mx 	5.4880
Mx 	7.5360
Mx 	5.5200
Mx 	7.1200
T	517 a=4.1920 607 a=5.4400 498 a=5.4880 567 a=7.4400 486 a=5.5200 527 a=7.1200	(french)
Mx 	115.0662
Mx 	6.5120
Mx 	5.1520
Mx 	6.3680
Mx 	6.3680
Mx 	4.1920
T	604 a=6.5120 607 a=5.4400 577 a=6.3680 577 a=6.3680 517 a=4.1920	(proof)
Mx 	145.9046
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2371 w=3.9520	(!)
Mx 	154.3666
T	2403 w=4.9440	(()
Mx 	160.0384
Set font 	Cormorant Infant;16;400;Italic;normal;;;LTR
T	718 555 634 816 w=23.7280	(fluff)
Mx 	185.8624
Set font 	Cormorant Infant;16;400;;normal;;;LTR
T	2404 w=4.9440	())
End page
Finish
